File: Plus One/Plus One.php
<?php

function plusOne($digits) {
    $new_digits = [];
    $carry = 1;
    for($i=count($digits)-1; $i>=0; $i--){
        $val_sum = $digits[$i]+$carry;
        if($val_sum>9){
            array_unshift($new_digits, 0);
            $carry = 1;
        }
        else{
            array_unshift($new_digits, $val_sum);
            $carry = 0;
        }
    }
    if($carry>0){
        array_unshift($new_digits, $carry);
    }

    return $new_digits;  
}

print(plusOne([1, 2, 3]));
?>

File: Plus One/Plus One.py
import collections

def plusOne(digits):
    """
    :type digits: List[int]
    :rtype: List[int]
    """
    new_digits = collections.deque()
    carry = 1
    for i in range(len(digits)-1, -1, -1):
        sum1 = digits[i]+carry
        if sum1>9:
            new_digits.appendleft(0)
            carry = 1
        else:
            new_digits.appendleft(sum1)
            carry = 0

    if carry>0:
        new_digits.appendleft(carry)

    return new_digits

print(plusOne([1,2,3]))